gnmi_cli_py/py_gnmicli.py

```diff
@@ -394,117 +423,142 @@ def gen_request(paths, opt, prefix):
-    elif response.notification[0].update[0].val.string_val:
-      print(response.notification[0].update[0].val.string_val)
-    else:
-      print('JSON Format specified, but gNMI Response was not json_ietf_val')
-      print(response)
-  elif mode == 'set-update':
-    print('Performing SetRequest Update, encoding=JSON_IETF', ' to ', target,
-          ' with the following gNMI Path\n', '-'*25, '\n', paths, json_value)
-    response = _set(stub, paths, 'update', user, password, json_value)
-    print('The SetRequest response is below\n' + '-'*25 + '\n', response)
-  elif mode == 'set-replace':
-    print('Performing SetRequest Replace, encoding=JSON_IETF', ' to ', target,
-          ' with the following gNMI Path\n', '-'*25, '\n', paths)
-    response = _set(stub, paths, 'replace', user, password, json_value)
-    print('The SetRequest response is below\n' + '-'*25 + '\n', response)
-  elif mode == 'set-delete':
-    print('Performing SetRequest Delete, encoding=JSON_IETF', ' to ', target,
-          ' with the following gNMI Path\n', '-'*25, '\n', paths)
-    response = _set(stub, paths, 'delete', user, password, json_value)
-    print('The SetRequest response is below\n' + '-'*25 + '\n', response)
-  elif mode == 'subscribe':
-    request_iterator = gen_request(paths, args, prefix)
-    subscribe_start(stub, args, request_iterator)
+    argparser = _create_parser()
+    args = vars(argparser.parse_args())
+    if args['version']:
+        print(__version__)
+        sys.exit()
+    if args['debug']:
+        os.environ['GRPC_TRACE'] = 'all'
+        os.environ['GRPC_VERBOSITY'] = 'DEBUG'
+    mode = args['mode']
+    target = args['target']
+    port = args['port']
+    notls = args['notls']
+    get_cert = args['get_cert']
+    root_cert = args['root_cert']
+    cert_chain = args['cert_chain']
+    json_value = args['value']
+    private_key = args['private_key']
+    xpath = args['xpath']
+    prefix = gnmi_pb2.Path(target=args['xpath_target'])
+    host_override = args['host_override']
+    user = args['username']
+    password = args['password']
+    form = args['format']
+    trigger_mem_spike = args['trigger_mem_spike']
+    paths = _parse_path(_path_names(xpath))
+    kwargs = {'root_cert': root_cert, 'cert_chain': cert_chain,
+              'private_key': private_key}
+    certs = _open_certs(**kwargs)
+    creds = _build_creds(target, port, get_cert, certs, notls)
+
+    if trigger_mem_spike:
+        while True:
+            try:
+                stub = _create_stub(creds, target, port, host_override)
+                request_iterator = gen_request(paths, args, prefix)
+                subscribe_start(stub, args, request_iterator)
+            except grpc.RpcError as err:
+                if err.code() == grpc.StatusCode.UNAVAILABLE:
+                    print("Receives an exception '{}' indicating gNMI server shuts down and exiting ..."
+                          .format(err.details()))
+                    sys.exit(1)
```

Comment:
@qiluo-msft The code snippet from 514 ~ 524 will simulate the behavior of gNMI C# client when it was invoked to poll the telemetry server in production devices.

Specifically, in the PR ( https://github.com/Azure/sonic-mgmt/pull/5296) of sonic-mgmt, test script (https://github.com/Azure/sonic-mgmt/blob/5e6e8cb3ac6434e693a15e677dab8d0c3774aaf5/tests/telemetry/test_telemetry.py#L282) will invoke this python gNMI client tool with the option `--trigger_mem_spike` to query the `valid/invalid` paths `DOCKER_STATS,TEST_STATS` in `STATE_DB`. Following is the pattern:

        1. telemetry server received initial query subscribe from Client_IP:port1
        2. Invalid db table Path STATE_DB DOCKER_STATS_TABLE,TEST_STATS_TABLE
        3. Client_IP:port1 shutdown
        4. telemetry server received initial query subscribe from Client_IP:port2
        5. Invalid db table Path STATE_DB DOCKER_STATS_TABLE,TEST_STATS_TABLE
        6. Client_IP:port2 shutdown
        7. Repeat from Step 1

In order to shutdown the gNMI client automatically after running sonic-mgmt test script, code from line 521 ~ 524 was added to detect whether telemetry server was available or not. If telemetry container was restarted, then this python gNMI client will catch such exception and exit automatically.